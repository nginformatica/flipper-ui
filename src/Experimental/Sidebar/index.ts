export { default as Sidebar } from './SidebarV2'
