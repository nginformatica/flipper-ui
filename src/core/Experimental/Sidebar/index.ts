export { default } from './SidebarV2'
