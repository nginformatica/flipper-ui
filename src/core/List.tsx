import {
    List as MuiList,
    ListSubheader as MuiListHeader
} from '@material-ui/core'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import { Theme } from '@material-ui/core/styles/createMuiTheme'
import React, { FC } from 'react'
import { DefaultProps } from './types'

interface ListProps extends DefaultProps {
    title?: string
    dense?: boolean
    color?: 'primary' | 'secondary' | 'default' | 'inherit'
}

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        default: {
            backgroundColor: theme.palette.background.default,
            color: theme.palette.text.primary
        },
        inherit: {
            backgroundColor: 'inherit',
            color: 'inherit'
        },
        primary: {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.primary.contrastText
        },
        secondary: {
            backgroundColor: theme.palette.secondary.main,
            color: theme.palette.secondary.contrastText
        }
    })
)

const List: FC<ListProps> = ({
    title,
    padding,
    margin,
    style = {},
    children,
    className,
    color = 'default',
    ...otherProps
}) => {
    const classes = useStyles()

    return (
        <MuiList
            subheader={
                title ? (
                    <MuiListHeader className={classes ? classes[color] : ''}>
                        {title}
                    </MuiListHeader>
                ) : undefined
            }
            className={classes ? `${classes[color]} ${className}` : ''}
            style={{ padding, margin, ...style }}
            {...otherProps}>
            {children}
        </MuiList>
    )
}

export default List
