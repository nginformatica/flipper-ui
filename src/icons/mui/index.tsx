export { default as IconAccessible } from '@mui/icons-material/Accessible'
export { default as IconAdd } from '@mui/icons-material/Add'
export { default as IconAccountCircle } from '@mui/icons-material/AccountCircle'
export { default as IconAddCircleOutline } from '@mui/icons-material/AddCircleOutline'
export { default as IconAllInbox } from '@mui/icons-material/AllInbox'
export { default as IconArrowBack } from '@mui/icons-material/ArrowBack'
export { default as IconArrowBackIos } from '@mui/icons-material/ArrowBackIos'
export { default as IconAssignment } from '@mui/icons-material/Assignment'
export { default as IconAssignmentInd } from '@mui/icons-material/AssignmentInd'
export { default as IconAudiotrack } from '@mui/icons-material/Audiotrack'
export { default as IconArrowDownward } from '@mui/icons-material/ArrowDownward'
export { default as IconArrowDropDown } from '@mui/icons-material/ArrowDropDown'
export { default as IconArrowUpward } from '@mui/icons-material/ArrowUpward'
export { default as IconBackHand } from '@mui/icons-material/BackHand'
export { default as IconBallot } from '@mui/icons-material/Ballot'
export { default as IconBiotechSharp } from '@mui/icons-material/BiotechSharp'
export { default as IconBlock } from '@mui/icons-material/Block'
export { default as IconBorderColor } from '@mui/icons-material/BorderColor'
export { default as IconBuild } from '@mui/icons-material/Build'
export { default as IconBusiness } from '@mui/icons-material/Business'
export { default as IconCancelOutlined } from '@mui/icons-material/CancelOutlined'
export { default as IconCached } from '@mui/icons-material/Cached'
export { default as IconCallReceived } from '@mui/icons-material/CallReceived'
export { default as IconCheck } from '@mui/icons-material/Check'
export { default as IconCheckBox } from '@mui/icons-material/CheckBox'
export { default as IconCheckCircle } from '@mui/icons-material/CheckCircle'
export { default as IconCheckCircleOutlined } from '@mui/icons-material/CheckCircleOutlined'
export { default as IconCircle } from '@mui/icons-material/Circle'
export { default as IconClose } from '@mui/icons-material/Close'
export { default as IconCloudDownload } from '@mui/icons-material/CloudDownload'
export { default as IconCompareArrows } from '@mui/icons-material/CompareArrows'
export { default as IconContentCopy } from '@mui/icons-material/ContentCopy'
export { default as IconContentPaste } from '@mui/icons-material/ContentPaste'
export { default as IconCreditCard } from '@mui/icons-material/CreditCard'
export { default as IconDateRange } from '@mui/icons-material/DateRange'
export { default as IconDelete } from '@mui/icons-material/Delete'
export { default as IconDescription } from '@mui/icons-material/Description'
export { default as IconDescriptionOutlined } from '@mui/icons-material/DescriptionOutlined'
export { default as IconDoneAll } from '@mui/icons-material/DoneAll'
export { default as IconDonutLarge } from '@mui/icons-material/DonutLarge'
export { default as IconDragHandle } from '@mui/icons-material/DragHandle'
export { default as IconDragIndicator } from '@mui/icons-material/DragIndicator'
export { default as IconEdit } from '@mui/icons-material/Edit'
export { default as IconEmail } from '@mui/icons-material/Email'
export { default as IconEnhancedEncryption } from '@mui/icons-material/EnhancedEncryption'
export { default as IconEqualizer } from '@mui/icons-material/Equalizer'
export { default as IconError } from '@mui/icons-material/Error'
export { default as IconErrorOutline } from '@mui/icons-material/ErrorOutline'
export { default as IconExitToApp } from '@mui/icons-material/ExitToApp'
export { default as IconExpand } from '@mui/icons-material/Expand'
export { default as IconExpandLess } from '@mui/icons-material/ExpandLess'
export { default as IconExpandMore } from '@mui/icons-material/ExpandMore'
export { default as IconFace } from '@mui/icons-material/Face'
export { default as IconFactCheck } from '@mui/icons-material/FactCheck'
export { default as IconFavorite } from '@mui/icons-material/Favorite'
export { default as IconFeedbackOutlined } from '@mui/icons-material/FeedbackOutlined'
export { default as IconFileCopy } from '@mui/icons-material/FileCopy'
export { default as IconFilterList } from '@mui/icons-material/FilterList'
export { default as IconFirstPage } from '@mui/icons-material/FirstPage'
export { default as IconFolderShared } from '@mui/icons-material/FolderShared'
export { default as IconFormatColorFill } from '@mui/icons-material/FormatColorFill'
export { default as IconFormatColorText } from '@mui/icons-material/FormatColorText'
export { default as IconFormatListBulleted } from '@mui/icons-material/FormatListBulleted'
export { default as IconForumOutlined } from '@mui/icons-material/ForumOutlined'
export { default as IconFullscreenRounded } from '@mui/icons-material/FullscreenRounded'
export { default as IconHeadset } from '@mui/icons-material/Headset'
export { default as IconHelp } from '@mui/icons-material/Help'
export { default as IconHelpOutline } from '@mui/icons-material/HelpOutline'
export { default as IconHistory } from '@mui/icons-material/History'
export { default as IconHome } from '@mui/icons-material/Home'
export { default as IconHomeWorkOutlined } from '@mui/icons-material/HomeWorkOutlined'
export { default as IconHourglassFull } from '@mui/icons-material/HourglassFull'
export { default as IconVote } from '@mui/icons-material/HowToVote'
export { default as IconImportantDevices } from '@mui/icons-material/ImportantDevices'
export { default as IconInfo } from '@mui/icons-material/Info'
export { default as IconInsertDriveFile } from '@mui/icons-material/InsertDriveFile'
export { default as IconLabelImportant } from '@mui/icons-material/LabelImportant'
export { default as IconLastPage } from '@mui/icons-material/LastPage'
export { default as IconLaunch } from '@mui/icons-material/Launch'
export { default as IconLibraryBooksOutlined } from '@mui/icons-material/LibraryBooksOutlined'
export { default as IconList } from '@mui/icons-material/List'
export { default as IconListAlt } from '@mui/icons-material/ListAlt'
export { default as IconLiveHelp } from '@mui/icons-material/LiveHelp'
export { default as IconLocalOffer } from '@mui/icons-material/LocalOffer'
export { default as IconLock } from '@mui/icons-material/Lock'
export { default as IconLockOpen } from '@mui/icons-material/LockOpen'
export { default as IconLockOutlined } from '@mui/icons-material/LockOutlined'
export { default as IconLocalHospital } from '@mui/icons-material/LocalHospital'
export { default as IconMenu } from '@mui/icons-material/Menu'
export { default as IconMonitorHeart } from '@mui/icons-material/MonitorHeart'
export { default as IconMoreHoriz } from '@mui/icons-material/MoreHoriz'
export { default as IconMoreVert } from '@mui/icons-material/MoreVert'
export { default as IconMovie } from '@mui/icons-material/Movie'
export { default as IconNavigateBefore } from '@mui/icons-material/NavigateBefore'
export { default as IconNavigateNext } from '@mui/icons-material/NavigateNext'
export { default as IconNoteAdd } from '@mui/icons-material/NoteAdd'
export { default as IconNotifications } from '@mui/icons-material/Notifications'
export { default as IconOndemandVideoOutlined } from '@mui/icons-material/OndemandVideoOutlined'
export { default as IconPeopleAlt } from '@mui/icons-material/PeopleAlt'
export { default as IconPerson } from '@mui/icons-material/Person'
export { default as IconPersonAdd } from '@mui/icons-material/PersonAdd'
export { default as IconPersonPin } from '@mui/icons-material/PersonPin'
export { default as IconPhoto } from '@mui/icons-material/Photo'
export { default as IconPhotoCamera } from '@mui/icons-material/PhotoCamera'
export { default as IconPin } from '@mui/icons-material/Pin'
export { default as IconPlayCircleFilled } from '@mui/icons-material/PlayCircleFilled'
export { default as IconPrint } from '@mui/icons-material/Print'
export { default as IconPriorityHigh } from '@mui/icons-material/PriorityHigh'
export { default as IconRecentActors } from '@mui/icons-material/RecentActors'
export { default as IconRemoveCircleOutline } from '@mui/icons-material/RemoveCircleOutline'
export { default as IconRepeat } from '@mui/icons-material/Repeat'
export { default as IconReplay } from '@mui/icons-material/Replay'
export { default as IconReportOutlined } from '@mui/icons-material/ReportOutlined'
export { default as IconRestore } from '@mui/icons-material/Restore'
export { default as IconRoom } from '@mui/icons-material/Room'
export { default as IconRoomOutlined } from '@mui/icons-material/RoomOutlined'
export { default as IconSave } from '@mui/icons-material/Save'
export { default as IconSaveAlt } from '@mui/icons-material/SaveAlt'
export { default as IconSchedule } from '@mui/icons-material/Schedule'
export { default as IconSearch } from '@mui/icons-material/Search'
export { default as IconSentimentNeutral } from '@mui/icons-material/SentimentNeutral'
export { default as IconSentimentSatisfied } from '@mui/icons-material/SentimentSatisfied'
export { default as IconSentimentSatisfiedAlt } from '@mui/icons-material/SentimentSatisfiedAlt'
export { default as IconSentimentDissatisfied } from '@mui/icons-material/SentimentDissatisfied'
export { default as IconSentimentVeryDissatisfied } from '@mui/icons-material/SentimentVeryDissatisfied'
export { default as IconSentimentVerySatisfied } from '@mui/icons-material/SentimentVerySatisfied'
export { default as IconSensors } from '@mui/icons-material/Sensors'
export { default as IconSettings } from '@mui/icons-material/Settings'
export { default as IconSort } from '@mui/icons-material/Sort'
export { default as IconStar } from '@mui/icons-material/Star'
export { default as IconStarBorder } from '@mui/icons-material/StarBorder'
export { default as IconSync } from '@mui/icons-material/Sync'
export { default as IconSyncProblem } from '@mui/icons-material/SyncProblem'
export { default as IconTextFields } from '@mui/icons-material/TextFields'
export { default as IconTimeline } from '@mui/icons-material/Timeline'
export { default as IconToday } from '@mui/icons-material/Today'
export { default as IconUpdate } from '@mui/icons-material/Update'
export { default as IconVisibility } from '@mui/icons-material/Visibility'
export { default as IconVisibilityOff } from '@mui/icons-material/VisibilityOff'
export { default as IconVpnKey } from '@mui/icons-material/VpnKey'
export { default as IconWarning } from '@mui/icons-material/Warning'
export { default as IconWaterDrop } from '@mui/icons-material/WaterDrop'
export { default as IconWifiOff } from '@mui/icons-material/WifiOff'
