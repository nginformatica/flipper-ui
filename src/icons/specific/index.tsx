export { IconAccident } from './IconAccident'
export { IconActionPlans } from './IconActionPlans'
export { IconAddDocument } from './IconAddDocument'
export { IconAET } from './IconAET'
export { IconAnwser } from './IconAnwser'
export { IconASO } from './IconASO'
export { IconBelt } from './IconBelt'
export { IconBillet } from './IconBillet'
export { IconBiometry } from './IconBiometry'
export { IconBoot } from './IconBoot'
export { IconBranch } from './IconBranch'
export { IconBrigades } from './IconBrigades'
export { IconCalendarDay } from './IconCalendarDay'
export { IconCalendarMonth } from './IconCalendarMonth'
export { IconCalendarWeek } from './IconCalendarWeek'
export { IconCancelRounded } from './IconCancelRounded'
export { IconCards } from './IconCards'
export { IconChangeEquipment } from './IconChangeEquipment'
export { IconCircle } from './IconCircle'
export { IconCompany } from './IconCompany'
export { IconComponent } from './IconComponent'
export { IconConfirmRounded } from './IconConfirmRounded'
export { IconCostCenter } from './IconCostCenter'
export { IconCoverall } from './IconCoverall'
export { IconDashboard } from './IconDashboard'
export { IconDistribution } from './IconDistribution'
export { IconDoctor } from './IconDoctor'
export { IconDoctorCalendar } from './IconDoctorCalendar'
export { IconDonator } from './IconDonator'
export { IconDoneChecked } from './IconDoneChecked'
export { IconDoneUnchecked } from './IconDoneUnchecked'
export { IconDownloadQRCode } from './IconDownloadQRCode'
export { IconEmployee } from './IconEmployee'
export { IconEnvironmentalReports } from './IconEnvironmentalReports'
export { IconEquipment } from './IconEquipment'
export { IconErrorStroke } from './IconErrorStroke'
export { IconESocial } from './IconESocial'
export { IconExtinguishingSystem } from './IconExtinguishingSystem'
export { IconFaceScan } from './IconFaceScan'
export { IconFemale } from './IconFemale'
export { IconFileInform } from './IconFileInform'
export { IconFluig } from './IconFluig'
export { IconFunil } from './IconFunil'
export { IconGeneral } from './IconGeneral'
export { IconGift } from './IconGift'
export { IconHealth } from './IconHealth'
export { IconHeight } from './IconHeight'
export { IconIMC } from './IconIMC'
export { IconIPE } from './IconIPE'
export { IconLocalization } from './IconLocalization'
export { IconLTCAT } from './IconLTCAT'
export { IconLTP } from './IconLTP'
export { IconMail } from './IconMail'
export { IconMale } from './IconMale'
export { IconMandates } from './IconMandates'
export { IconMap } from './IconMap'
export { IconMask } from './IconMask'
export { IconMaterial } from './IconMaterial'
export { IconMedicalRecords } from './IconMedicalRecords'
export { IconMediumCompany } from './IconMediumCompany'
export { IconMegaCompany } from './IconMegaCompany'
export { IconMinimumCompany } from './IconMinimumCompany'
export { IconNickName } from './IconNickName'
export { IconOccupation } from './IconOccupation'
export { IconPAE } from './IconPAE'
export { IconPCMAT } from './IconPCMAT'
export { IconPCMSO } from './IconPCMSO'
export { IconPGR } from './IconPGR'
export { IconPGRTR } from './IconPGRTR'
export { IconPPR } from './IconPPR'
export { IconPPRA } from './IconPPRA'
export { IconQuestion } from './IconQuestion'
export { IconReturn } from './IconReturn'
export { IconRisks } from './IconRisks'
export { IconRotate } from './IconRotate'
export { IconSafetyGlasses } from './IconSafetyGlasses'
export { IconScheduleDashboard } from './IconScheduleDashboard'
export { IconScheduleDashboardHour } from './IconScheduleDashboardHour'
export { IconScheduleHour } from './IconScheduleHour'
export { IconServiceOrder } from './IconServiceOrder'
export { IconServiceRequest } from './IconServiceRequest'
export { IconSmallCompany } from './IconSmallCompany'
export { IconSpeciality } from './IconSpeciality'
export { IconSuperCompany } from './IconSuperCompany'
export { IconSupplier } from './IconSupplier'
export { IconThirdParty } from './IconThirdParty'
export { IconTree } from './IconTree'
export { IconTriangleDown } from './IconTriangleDown'
export { IconTriangleUp } from './IconTriangleUp'
export { IconVest } from './IconVest'
export { IconWarehouse } from './IconWarehouse'
export { IconWarningSpecific } from './IconWarningSpecific'
export { IconWeight } from './IconWeight'
export { IconWeldingMask } from './IconWeldingMask'
export { IconWindows } from './IconWindows'
export { IconWorkPermission } from './IconWorkPermission'
export { IconYellowPending } from './IconYellowPending'
