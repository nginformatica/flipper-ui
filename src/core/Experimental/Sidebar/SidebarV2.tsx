import FlipperSidebar from '../../Sidebar'
import React, { useEffect, useState, useCallback } from 'react'
import styled from 'styled-components'
import { theme } from 'nginformatica-styleguide'
import { times, map } from 'ramda'
import { Skeleton } from '@material-ui/lab'
import Tooltip from '../../Tooltip'
import List from '../../List'
import ListItem from '../../ListItem'

export interface ISidebarOption {
    icon: React.ReactElement
    label: string
    name: string
    route: string
}

export interface IProps {
    options: ISidebarOption[]
    top?: number
    loading?: boolean
    handleGoTo?: (name: string, route: string) => () => void
    extraOptions?: ISidebarOption[]
}

const StyledFlipperSidebar = styled(FlipperSidebar)`
    & > div {
        border-right: 1px solid ${theme.colors.grays.g6};
    }
`

const MenuItem = styled(ListItem)`
    &&& {
        & div {
            padding: 0;
        }
    }
`

const Link = styled.a`
    color: ${theme.colors.primary.main};
    text-decoration: none;
`

const PaperBar = styled.div`
    flex-direction: column;
    display: flex;
    justify-content: space-between;
    flex: 1;
    padding-bottom: 16px;
`

const LABEL_TIMEOUT = 300

const renderSkeleton = (index: number) => (
    <div
        className='skeleton-container'
        role='skeleton'
        key={`sidebar-skeleton-${index}`}
        style={{ padding: '2px 10px' }}>
        <Skeleton width={36} height={36} />
    </div>
)

const SIDEBAR_SKELETON_SIZE = 6
const sidebarSkeleton = times(renderSkeleton, SIDEBAR_SKELETON_SIZE)

// create a function that extract path from location.pathname
// and return the route
const extract = (location: string): string => {
    const lastSlashIndex = location.lastIndexOf('/')

    return '/' + location.slice(lastSlashIndex + 1)
}
const useSidebar = () => {
    const [expanded, setExpanded] = useState(false)
    const route = extract(location.pathname)

    const toggle = useCallback(() => {
        setExpanded(!expanded)
    }, [expanded])

    return { toggle, expanded, route }
}

const emptyList: ISidebarOption[] = []

/**
 * Sidebar component V2
 */
const Sidebar = (props: IProps) => {
    const {
        options,
        loading,
        handleGoTo,
        extraOptions = emptyList,
        top,
        ...otherProps
    } = props

    const { expanded, route, toggle } = useSidebar()

    const [showLabel, setShowLabel] = useState(expanded)

    useEffect(() => {
        if (expanded) {
            setTimeout(() => setShowLabel(true), LABEL_TIMEOUT)
        } else {
            setShowLabel(false)
        }
    }, [expanded])

    const renderOption = (option: ISidebarOption) => (
        <Link
            data-testid={`sidebar-option-${option.route.replace('/', '')}`}
            key={option.name}>
            <Tooltip
                withWrapper
                placement='right'
                enterDelay={500}
                title={expanded ? '' : option.label}>
                <MenuItem
                    key={option.name}
                    selected={option.name === route}
                    icon={option.icon}
                    style={{ minHeight: '42px' }}
                    padding='0 18px'
                    subtitle={showLabel ? option.label : ''}
                    onClick={handleGoTo?.(option.name, option.route)}
                />
            </Tooltip>
        </Link>
    )

    return (
        <StyledFlipperSidebar
            open
            docked
            name='sidebar'
            id='sidebar'
            ButtonProps={{ id: 'sidebar-toggle-button' }}
            minWidth={56}
            maxWidth={228}
            style={{ transition: 'width 100ms ease-in-out', zIndex: 1199 }}
            expanded={expanded}
            onToggle={toggle}
            top={top || 0}>
            <PaperBar {...otherProps}>
                <List dense>
                    {loading ? sidebarSkeleton : map(renderOption, options)}
                </List>
                <List dense>{!loading && map(renderOption, extraOptions)}</List>
            </PaperBar>
        </StyledFlipperSidebar>
    )
}

export default Sidebar
