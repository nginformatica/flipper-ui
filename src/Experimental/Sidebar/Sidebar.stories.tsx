import React from 'react'
import Sidebar from '.'
import { Home, Computer, Edit } from '@material-ui/icons'
import { Meta, StoryFn } from '@storybook/react'

export default {
    title: 'experimental/Sidebar',
    component: Sidebar
} as Meta<typeof Sidebar>

const Template: StoryFn<typeof Sidebar> = args => <Sidebar {...args} />

export const Default = Template.bind({})
Default.args = {
    options: [
        {
            icon: <Home />,
            name: 'Home',
            label: 'Home',
            route: '/'
        },
        {
            icon: <Computer />,
            name: 'Computer',
            label: 'Computer',
            route: '/computer'
        }
    ],
    extraOptions: [
        {
            icon: <Edit />,
            name: 'Edit',
            label: 'edit',
            route: '/edit'
        }
    ],
    handleGoTo: (foo: string, bar: string) => () => {
        alert(foo + ' ' + bar)
    }
}

export const Loading = Template.bind({})
Loading.args = {
    loading: true,
    options: []
}
