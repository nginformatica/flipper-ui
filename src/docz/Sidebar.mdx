---
name: Sidebar
route: /sidebar
---

import Sidebar from '../core/Sidebar'
import Container from '../core/Container'
import List from '../core/List'
import ListItem from '../core/ListItem'
import { Backup as IconBackup } from '../icons'
import { Playground } from 'docz'

# Sidebar

## Normal
<Playground>
    <Container
        style={ { height: 256 } }>
        <Sidebar expanded={ false }>
            <List>
                <ListItem icon={ <IconBackup /> } />
                <ListItem icon={ <IconBackup /> } />
            </List>
        </Sidebar>
    </Container>
</Playground>

## Right Sidebar
<Playground>
    <Container
        style={ {
            margin: '24px',
            flexDirection: 'row-reverse',
            height: 256
        } }>
        <Sidebar open anchor='right'>
            <List>
                <ListItem icon={ <IconBackup /> } title='Item 1' />
                <ListItem icon={ <IconBackup /> } title='Item 2' />
            </List>
        </Sidebar>
    </Container>
</Playground>

## Opened Sidebar
<Playground>
    <Container style={ { height: 256 } }>
        <Sidebar open>
            <List>
                <ListItem icon={ <IconBackup /> } title='Item 1' />
                <ListItem icon={ <IconBackup /> } title='Item 2' />
            </List>
        </Sidebar>
    </Container>
</Playground>
