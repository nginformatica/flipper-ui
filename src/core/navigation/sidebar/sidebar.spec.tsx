import * as React from 'react'
import { render, screen } from '@testing-library/react'
import List from '@/core/data-display/list'
import ListItem from '@/core/data-display/list-item'
import Sidebar from '.'
import { Backup } from '@/icons'

describe('Sidebar', () => {
    it('should render', () => {
        render(
            <Sidebar open onToggle={jest.fn()}>
                <List>
                    <ListItem title='Item 1' icon={<Backup />} />
                    <ListItem title='Item 2' icon={<Backup />} />
                </List>
            </Sidebar>
        )

        const item1 = screen.getByText('Item 1')
        const item2 = screen.getByText('Item 2')

        expect(item1).toBeDefined()
        expect(item2).toBeDefined()
    })

    it('should render closed', () => {
        render(
            <Sidebar open={false} onToggle={jest.fn()}>
                <List>
                    <ListItem title='Item 1' icon={<Backup />} />
                    <ListItem title='Item 2' icon={<Backup />} />
                </List>
            </Sidebar>
        )

        const item1 = screen.getByText('Item 1')
        const item2 = screen.getByText('Item 2')

        expect(item1).toBeDefined()
        expect(item2).toBeDefined()
    })

    it('should render unExpanded', () => {
        render(
            <Sidebar open expanded={false} onToggle={jest.fn()}>
                <List>
                    <ListItem title='Item 1' icon={<Backup />} />
                    <ListItem title='Item 2' icon={<Backup />} />
                </List>
            </Sidebar>
        )

        const item1 = screen.getByText('Item 1')
        const item2 = screen.getByText('Item 2')

        expect(item1).toBeDefined()
        expect(item2).toBeDefined()
    })

    it('should render with anchor right', () => {
        render(
            <Sidebar open anchor='right' onToggle={jest.fn()}>
                <List>
                    <ListItem title='Item 1' icon={<Backup />} />
                    <ListItem title='Item 2' icon={<Backup />} />
                </List>
            </Sidebar>
        )

        const item1 = screen.getByText('Item 1')
        const item2 = screen.getByText('Item 2')

        expect(item1).toBeDefined()
        expect(item2).toBeDefined()
    })
})
