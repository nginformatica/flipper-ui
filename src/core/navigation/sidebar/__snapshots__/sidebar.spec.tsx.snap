// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked"
    style="width: 220px; top: 64px;"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft  makeStyles-default-54 MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        anchor="left"
        class="sc-aYaIB eVWnkT"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-53"
          data-testid="sidebar-button"
          name="button-sidebar"
          style="max-width: 72px; opacity: 1;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-55 css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <ul
        class="MuiList-root makeStyles-default-60 undefined MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-64 MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-default-65"
            style="min-width: 42px;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="BackupIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M14 13v4h-4v-4H7l5-5 5 5z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
            data-testid="list-item-undefined"
          >
            <span
              class="MuiTypography-root makeStyles-default-65 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Item 1
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-root-64 MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-default-65"
            style="min-width: 42px;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="BackupIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M14 13v4h-4v-4H7l5-5 5 5z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
            data-testid="list-item-undefined"
          >
            <span
              class="MuiTypography-root makeStyles-default-65 MuiTypography-body1 MuiTypography-displayBlock"
            >
              Item 2
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
